plugin.video.hrmagplayer/libplayer/player.py

```diff
@@ -0,0 +1,134 @@
+        print logStr
+        xbmc.log(logStr, xbmc.LOGERROR)
+    
+def list_shows(context):
+    """
+    Create the list of video categories in the Kodi interface.
+    :return: None
+    """
+    # Get video categories
+    shows = context['showList']
+    # Create a list for our items.
+    listing = []
+    # Iterate through categories
+    index = 0
+    for show in shows:
+        if show['active']:
+            # Create a list item with a text label and a thumbnail image.
+            list_item = xbmcgui.ListItem(label=show['name'], thumbnailImage=show['image'])
+ 
+            # Create a URL for the plugin recursive callback.
+            # Example: plugin://plugin.video.example/?action=listing&category=Animals
+            is_folder = True
+            if index == 0:
+                is_folder = False
+            # Pass 'listing' and show index
+            url = '{0}?action=listing&show={1}'.format(__url__, str(index))
+            # is_folder = True means that this item opens a sub-list of lower level items.
+
+            # Add our item to the listing as a 3-element tuple.
+            listing.append((url, list_item, is_folder))
+        index += 1
+    # Add our listing to Kodi.
+    # Large lists and/or slower systems benefit from adding all items at once via addDirectoryItems
+    # instead of adding one by ove via addDirectoryItem.
+    xbmcplugin.addDirectoryItems(__handle__, listing, len(listing))
+    # Add a sort method for the virtual folder items (alphabetically, ignore articles)
+    #xbmcplugin.addSortMethod(__handle__, xbmcplugin.SORT_METHOD_LABEL_IGNORE_THE)
+    # Finish creating a virtual folder.
+    xbmcplugin.endOfDirectory(__handle__)
+    
+def list_episodes(context, show):
+    # Get video categories
+    showid = context['showList']
+    episodes = context['episodes']
+    # Create a list for our items.
+    listing = []
+    # Iterate through categories
+    index = 0
+    for episode in episodes:
+        # Create a list item with a text label and a thumbnail image.
+        list_item = xbmcgui.ListItem(label=episode['title'], thumbnailImage=episode['image'])
+        #list_item.setInfo('video', 'Test Info')
+        url = '{0}?action=play&url={1}&episode={2}&show={3}'.format(__url__, episode['link'], encode(episode['title']), show)
+        is_folder = False
+        listing.append((url, list_item, is_folder))
+        index += 1
+    xbmcplugin.addDirectoryItems(__handle__, listing, len(listing))
+    xbmcplugin.endOfDirectory(__handle__)
+    
+def dispatch(url, handle, parameter):
+    logStr = "Entry: " + url + ", " + str(handle) + ", " + parameter
+    xbmc.log(logStr, xbmc.LOGINFO)
+    parameters = extractParameters(parameter)
+    action = None
+    show = None
+    url = None
+    episode = None
+    if parameters != None:
+        for action in parameters:
+            logStr = action + " = " + parameters[action]
+            xbmc.log(logStr, xbmc.LOGINFO)
```

Comment:
careful with log flooding (I believe there are cases in which you can simply use the DEBUG loglevel instead) that will only be shown if the user has debug logging enabled